File: php/memory.php
<?php
    include "hardware.php";

    class memory extends hardware{
        // atribut
        private $frequency = 0;
        private $memorySize = 0;
        private $supportCuda = "blank";

        // constructor
        function __construct(){}

        // Setter dan Getter

        function setFrequency($frequency){
            $this->frequency = $frequency;
        }
        function getFrequency(){
            return $this->frequency;
        }

        function setMemorySize($memorySize){
            $this->memorySize = $memorySize;
        }
        function getMemorySize(){
            return $this->memorySize;
        }

        function setSupportCuda($supportCuda){
            $this->supportCuda = $supportCuda;
        }
        function getSupportCuda(){
            return $this->supportCuda;
        }

        // destructor
        function __destruct(){}

    }
?>

File: latihan2/java/memory.java
class memory extends hardware {
    // atribut
    private int frequency;
    private int memorySize;
    private String supportsCuda;

    // constructor kosong
    memory(){}

    // Setter dan Getter

    public int getFrequency() {
        return frequency;
    }
    public void setFrequency(int frequency) {
        this.frequency = frequency;
    }
    public int getMemorySize() {
        return memorySize;
    }
    public void setMemorySize(int memorySize) {
        this.memorySize = memorySize;
    }
    public String getSupportsCuda() {
        return supportsCuda;
    }
    public void setSupportsCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }

}